File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (nums, target) {
    const ans = new Map();
    ans.set(nums[0], 0);
    for (let i = 1; i < nums.length; i++) {
        if (ans.has(target - nums[i])) {
            return [i, ans.get(target - nums[i])];
        }
        ans.set(nums[i], i)
    }
    return [];
};

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        if (nums.length == 2) {
            return new int[] { 0, 1 };
        }

        Map<Integer, Integer> diffs = new HashMap<Integer, Integer>();
        for (int i = 0; i < nums.length; i++) {
            if (diffs.containsKey(target - nums[i])) {
                return new int[] { diffs.get((target - nums[i])), i };
            }
            diffs.put(nums[i], i);
        }
        System.out.println(diffs);
        return new int[] { 0, 1 };
    }
}